django_cas_ng/backends.py

```diff
@@ -26,11 +26,15 @@ def authenticate(self, ticket, service, request):
         if not username:
             return None
 
-        username_case = settings.CAS_FORCE_CHANGE_USERNAME_CASE
-        if username_case == 'lower':
-            username = username.lower()
-        elif username_case == 'upper':
-            username = username.upper()
+        for func in settings.CAS_USERNAME_NORMALIZATIONS:
+            if callable(func):
+                username = func(username)
+            elif func == 'lower':
+                username = username.lower()
+            elif func == 'upper':
+                username = username.upper()
+            elif func == 'strip':
+                username = username.strip()
 
```

Comment:
Agreed. I did not add validation here because there was none in the existing implementation of `CAS_FORCE_CHANGE_USERNAME_CASE` did not have validation either.

However after thinking about this more I now believe that this patches introduces more complexity than should be acceptable from a setting, I will provide a different patch that allows for customization of this through a backend method the same way [Django does for the RemoteUserBackend](https://docs.djangoproject.com/en/1.10/ref/contrib/auth/#django.contrib.auth.backends.RemoteUserBackend.clean_username).